src/node/app/login.ts

```diff
@@ -108,3 +115,18 @@ export class LoginHttpProvider extends HttpProvider {
     throw new Error("Missing password")
   }
 }
+
+// RateLimiter wraps around the limiter library for logins.
+// It allows 2 logins every minute and 12 logins every hour.
+class RateLimiter {
+  private readonly minuteLimiter = new limiter.RateLimiter(2, "minute")
+  private readonly hourLimiter = new limiter.RateLimiter(12, "hour")
+
+  public try(): boolean {
+    console.info(this.minuteLimiter.getTokensRemaining(), this.hourLimiter.getTokensRemaining())
```

Comment:
Was this just for debugging? If not I think we might want to add some context. Although would it be a valid concern that it could be used to flood code-server's logs? Could put it behind a `debug` log instead I suppose.

Reply:
Yea will remove sorry.